### PR TITLE
Some readability improvements for xwayland server

### PR DESCRIPTION
**Pull request by JassonCordones**
#### Describe your PR, what does it fix/add?
Some readability improvements for xwayland server.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
No

#### Is it ready for merging, or does it need work?
Yes. Unless something else is required.

### CODE REVIEW

File: src/xwayland/Server.cpp [hunk 1]
```diff
@@ -21,124 +18,156 @@
 #include <filesystem>
 #include <cstring>
 
-// TODO: cleanup
-static bool set_cloexec(int fd, bool cloexec) {
+// Project includes
```
Comment (vaxerski) [inline src/xwayland/Server.cpp:21]: these comments are redundant

File: src/xwayland/Server.cpp [hunk 2]
```diff
@@ -21,124 +18,156 @@
...
+
     return true;
 }
 
-static int openSocket(struct sockaddr_un* addr, size_t path_size) {
-    int       fd, rc;
-    socklen_t size = offsetof(struct sockaddr_un, sun_path) + path_size + 1;
+void cleanUpSocket(int fd, const char* path) {
+    close(fd);
+    if (path[0]) {
```
Comment (vaxerski) [inline src/xwayland/Server.cpp:60]: no {}

File: src/xwayland/Server.cpp [hunk 3]
```diff
@@ -21,124 +18,156 @@
...
-static bool openSockets(std::array<int, 2>& sockets, int display) {
-    auto ret = mkdir("/tmp/.X11-unix", 755);
-
-    if (ret != 0) {
-        if (errno == EEXIST) {
-            if (!checkPermissionsForSocketDir())
-                return false;
+static bool ensureSocketDirExists(){
+    if(mkdir("/tmp/.X11-unix", SOCKET_DIR_PERMISSIONS)!=0){
+        if(errno == EEXIST){
```
Comment (vaxerski) [inline src/xwayland/Server.cpp:127]: no {}

File: src/xwayland/Server.cpp [hunk 4]
```diff
@@ -21,124 +18,156 @@
...
-            Debug::log(ERR, "XWayland: couldn't create socket dir");
+            Debug::log(ERR, "XWayland: Couldn't create socket dir");
             return false;
         }
     }
+    return true;
+}
+
+static std::string getSocketPath(int display, bool isLinux){
+    if(isLinux){
```
Comment (vaxerski) [inline src/xwayland/Server.cpp:138]: you get me

**Review by vaxerski — COMMENTED** [review]
also in dire need of a clang-format

File: src/xwayland/Server.cpp (hunk 3 above)
Comment (JassonCordones) [reply src/xwayland/Server.cpp]: I left this one. I was getting a dangling else warning if I removed the {}

Comment (vaxerski) [reply src/xwayland/Server.cpp]: the `errno` one, not the parent

Comment (JassonCordones) [reply src/xwayland/Server.cpp]: done

File: src/xwayland/Server.cpp [hunk 5]
```diff
@@ -203,19 +225,18 @@ bool CXWaylandServer::tryOpenSockets() {
 
         uint64_t pid = 0;
         try {
-            pid = std::stoi(std::string{pidstr, 11});
+            pid = std::stoul(pidstr);
```
Comment (vaxerski) [inline src/xwayland/Server.cpp:228]: ull? isnt pid a u64 on modern systems?

**Review by vaxerski — COMMENTED** [review]
while you are at it, you could also remove all the other {} around short ifs

File: src/xwayland/Server.cpp (hunk 5 above)
Comment (leiserfg) [reply src/xwayland/Server.cpp:228]: `pid_t` in linux is an `int` 
 https://ftp.gnu.org/old-gnu/Manuals/glibc-2.2.3/html_node/libc_554.html

Comment (vaxerski) [reply src/xwayland/Server.cpp]: my bad

**Review by vaxerski — APPROVED** [review]
thanks!

**Comment by vaxerski** [comment]
oops clang-format needed

**Review by vaxerski — APPROVED** [review]

**Pull request merged**